code/drasil-gen/lib/Language/Drasil/Generate.hs

```diff
@@ -89,12 +105,27 @@ prntCSS docType fn body = do
   where
     getFD dtype = show dtype ++ "/HTML/"
 
--- | Renders the documents.
+-- | Helper for generating the .toml config file for mdBook.
+printBook :: DocType -> Document -> PrintingInformation -> IO()
+printBook dt doc sm = do
+  outh <- openFile fp WriteMode
+  hPutStrLn outh $ render (makeBook doc sm)
+  hClose outh
+  where
+    fp = show dt ++ "/mdBook" ++ "/book.toml"
+
+-- | Renders single-page documents.
 writeDoc :: PrintingInformation -> DocType -> Format -> Filename -> Document -> Doc
-writeDoc s _  TeX  _  doc = genTeX doc s
-writeDoc s _  HTML fn doc = genHTML s fn doc
-writeDoc s dt JSON _  doc = genJSON s dt doc
-writeDoc _ _  _    _  _   = error "we can only write TeX/HTML/JSON (for now)"
+writeDoc s _  TeX      _  doc = genTeX doc s
+writeDoc s _  HTML     fn doc = genHTML s fn doc
+writeDoc s dt JSON     _  doc = genJSON s dt doc
+writeDoc s _  Markdown _  doc = genMD s doc
+writeDoc _ _  _        _  _   = error "we can only write TeX/HTML/JSON/Markdown/MDBook (for now)"
+
+-- | Renders multi-page documents.
+writeDoc' :: PrintingInformation -> Format -> Document -> [(String, Doc)]
+writeDoc' s MDBook doc = genMDBook s doc
+writeDoc' _ _      _   = error "we can only write TeX/HTML/JSON/Markdown/MDBook (for now)"
```

Comment:
This error string should be made consistent and reused instead of manually defining it repeatedly

Reply:
Fixed in https://github.com/JacquesCarette/Drasil/pull/3823/commits/f7ba8fda90132e900e5349385606160f9c09ac72.